virttest/storage.py

```diff
@@ -618,7 +619,31 @@ def __init__(self, params, root_dir, tag):
             self.snapshot_format = ss_params.get("image_format")
 
         self.image_access = ImageAccessInfo.access_info_define_by_params(
-                                                   self.tag, self.params)
+            self.tag, self.params)
+
+        self.data_file = self.external_data_file_defined_by_params(
+            params, root_dir, tag)
+
+    @classmethod
+    def external_data_file_defined_by_params(cls, params, root_dir, tag):
+        """Link image to an external data file."""
+        enable_data_file = params.get("enable_data_file", "no") == "yes"
+        image_format = params.get("image_format", "qcow2")
+        if image_format != "qcow2" or not enable_data_file:
+            return
```

Comment:
Should raise `ValueError` for those formats don't support external data file.

Reply:
update, thanks